pkg/kvstore/redis.go

```diff
@@ -162,13 +169,50 @@ func (s *redisKvStore) PutBatch(ctx context.Context, values interface{}) error {
 		return fmt.Errorf("could not convert values from %T to map[interface{}]interface{}", values)
 	}
 
+	chunkSize := s.settings.BatchSize
+	pairs := make([]interface{}, 0, chunkSize)
 	for k, v := range mii {
-		if err = s.Put(ctx, k, v); err != nil {
-			return fmt.Errorf("failed to write batch to redis: %w", err)
+		key, value, err := s.MarshalKeyValue(k, v)
+		if err != nil {
+			return fmt.Errorf("PutBatch could not marshal key/value: %w", err)
+		}
+		pairs = append(pairs, key)
+		pairs = append(pairs, value)
+
+		if len(pairs) >= 2*chunkSize {
+			err = s.flushChunk(ctx, pairs)
+			if err != nil {
+				return fmt.Errorf("failed to write batch to redis: %w", err)
+			}
+			pairs = make([]interface{}, 0, chunkSize)
 		}
 	}
 
-	return nil
+	return s.flushChunk(ctx, pairs)
+}
+
+func (s *redisKvStore) flushChunk(ctx context.Context, pairs []interface{}) error {
+	if len(pairs) < 1 {
+		return nil
+	}
+
+	err := s.client.MSet(ctx, pairs)
```

Comment:
Hmm, can we also do the mset in the pipeline? Is Pipeline the go equivalent of `MULTI` in redis? Because otherwise we have a strange state visible where the data was set, but has no ttl set.

Reply:
umm, pipelining by itself, is just saving on network roundtrip, so no transaction behaviour,
but we can easily make the whole thing a transaction.
on it.

Reply:
done